hathor/indexes/rocksdb_tokens_index.py

```diff
@@ -0,0 +1,463 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+from dataclasses import dataclass
+from enum import Enum
+from typing import TYPE_CHECKING, Iterator, List, NamedTuple, Optional, Tuple, cast
+
+from structlog import get_logger
+
+from hathor.conf import HathorSettings
+from hathor.indexes.rocksdb_utils import RocksDBIndexUtils
+from hathor.indexes.tokens_index import TokenIndexInfo, TokensIndex, TokenUtxoInfo
+from hathor.transaction import BaseTransaction, Transaction
+from hathor.transaction.base_transaction import TxVersion
+from hathor.util import collect_n, json_dumpb, json_loadb
+
+# TypedDict not available on Python 3.7, but is available from typing_extensions
+if sys.version_info < (3,) or sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+if TYPE_CHECKING:  # pragma: no cover
+    import rocksdb
+
+settings = HathorSettings()
+logger = get_logger()
+
+_CF_NAME_TOKENS_INDEX = b'tokens-index'
+_INTERNAL_HATHOR_TOKEN_UID = b'\x00' * 32
```

Comment:
I'd rather leave it here, this is completely internal to the implementation. Instead of storing `settings.HATHOR_TOKEN_UID`, we store a 32-byte sequence of zeros, just so that it can have a fixed size.

Reply:
Yeah, I agree with you, it makes sense.

Reply:
But doesn't it make sense to use?
```python
_INTERNAL_HATHOR_TOKEN_UID = settings.HATHOR_TOKEN_UID * 32
```

Reply:
> But doesn't it make sense to use?
> 
> ```python
> _INTERNAL_HATHOR_TOKEN_UID = settings.HATHOR_TOKEN_UID * 32
> ```

Only _if_ `len(settings.HATHOR_TOKEN_UID ) == 1` and maybe also `settings.HATHOR_TOKEN_UID == b'\x00'`.

I don't think there is any benefit of doing that here because I really just mean to use a 32-byte sequence of zeros that will be translated to whatever `settings.HATHOR_TOKEN_UID` is.

Reply:
Got it.